src/components/CandidateTaskView/CandidateTaskView.js

```diff
@@ -1,41 +1,107 @@
 import { observer } from "mobx-react";
-import React from "react";
+import React, { forwardRef, useRef } from "react";
 import { Block, Elem } from "../../utils/bem";
 import "./CandidateTaskView.styl";
+import { getRoot } from "mobx-state-tree";
+import { useEffect, useState } from "react";
+import { format } from "date-fns";
 
-const DataItemVisual = ({ columns, dataKey, data }) => {
+const DataItemVisual = forwardRef(({ columns, dataKey, data }, imageRef) => {
   const columnDefinition = columns.find(colData => colData.alias === dataKey);
 
   if (columnDefinition?.currentType === "Image") {
     return (
-      <Elem name='data-display' mod={{ image: true }}>
-        <img src={data} />
+      <Elem name="data-display" mod={{ image: true }}>
+        <img ref={imageRef} src={data} />
       </Elem>
     );
   }
   return (
-    <Elem name='data-display' mod={{ text: true }} >
+    <Elem name="data-display" mod={{ text: true }} >
       {data}
     </Elem>
   );
-};
+});
+
+const AttributeRow = (({ fieldName, value }) => {
+  return (
+    <Block name='attributeRow'>
+      <Elem name='name'>{fieldName}</Elem>
+      <Elem name='value'>{value}</Elem>
+    </Block>
+  );
+});
+const dateDisplayFormat = "MMM dd, yyyy HH:mm a";
 
 export const CandidateTaskView = observer(({ item, columns }) => {
-  const { data } = item;
+  const { candidate_task_id, id, data } = item;
+  const dataset = getRoot(item)?.SDK?.dataset;
+  const [fName, setFName] = useState();
+  const [fType, setFType] = useState();
+  const [mType, setMType] = useState();
+  const [created, setCreated] = useState();
+  const [modified, setModified] = useState();
+  const [size, setSize] = useState();
+  const [dimensions, setDimensions] = useState([]);
+  const [bucket, setBucket] = useState();
+  const imgRef = useRef({});
+
+  useEffect(async () => {
+    const { metadata } = await getRoot(item).apiCall("candidateTaskMeta", {
+      candidate_task_id,
+    });
+
+    if (metadata) {
+      setFName(metadata.name.split('/').pop());
+      setFType(metadata.contentType.split('/').shift());
+      setMType(metadata.contentType);
+      setCreated(metadata.timeCreated ? format(new Date(metadata.timeCreated), dateDisplayFormat) : "");
+      setModified(metadata.updated ? format(new Date(metadata.updated), dateDisplayFormat) : "");
+      setSize(`${new Intl.NumberFormat().format(parseInt(metadata.size))} bytes`);
+      setBucket(metadata.bucket);
+      setDimensions(Object.values(imgRef.current).map(ref => `${ref.naturalWidth} x ${ref.naturalHeight} px`));
+    }
+  }, [candidate_task_id]);
+  
```

Comment:
`useEffect` is async by its nature, making a callback async can lead to unpredicted behavior. you also lose ability to return a cleanup function as your callback returns promise instead

it's better to define an async function inside and call it right away

```suggestion
  useEffect(() => {
    const setDefaultMetadata = async () => {
      const { metadata } = await getRoot(item).apiCall("candidateTaskMeta", {
        candidate_task_id,
      });
  
      if (metadata) {
        setFName(metadata.name.split('/').pop());
        setFType(metadata.contentType.split('/').shift());
        setMType(metadata.contentType);
        setCreated(metadata.timeCreated ? format(new Date(metadata.timeCreated), dateDisplayFormat) : "");
        setModified(metadata.updated ? format(new Date(metadata.updated), dateDisplayFormat) : "");
        setSize(`${new Intl.NumberFormat().format(parseInt(metadata.size))} bytes`);
        setBucket(metadata.bucket);
        setDimensions(Object.values(imgRef.current).map(ref => `${ref.naturalWidth} x ${ref.naturalHeight} px`));
      }
    };
    
    setDefaultMetadata();
  }, [candidate_task_id]);
  
```